### DISCUSSION TITLE
Does Paste change APIs with minor version bumps?

### DISCUSSION THREAD

**Discussion by victoriahu** [twilio-labs/paste#1735] [General] [answered]
Can I confirm with the team that Paste doesn't change APIs with minor version bumps?

We are working on a dependency updating strategy for Console. Changing APIs in minor version bumps would mean we could only auto-update patch versions, which would make it harder to keep Console up-to-date on Paste.

Ideally, Console would always be up-to-date with Paste, so we need a contractual commitment from the team that APIs don't change with minor version bumps to help make that happen.

**Comment by SiTaggart** [comment, marked as answer]
Hi @victoriahu 

Correct, we strive to make sure patch and minor versions are non-impacting. We follow Semantic Versioning as best as we possibly can, and are becoming stricter with what we call a "breaking change" as our appetite for risk becomes smaller.

Major version bumps would include anything where the consumer needs to "do anything", but might also include things that have significant enough visual changes that you should upgrade with care, even if you don't need to change any code. 

For Core ,things we might consider major include:

- API changes
- Removal of exported code
- Non-backwards compatible typing changes. (making something required, significant shape change)
- Visual changes that might cause unintended visual side effects

For individual packages, like `@twilio-paste/button`, we consider a major change to include any of the above, plus:

- Adding a new peer dependency

Of course, we are human, and we do make mistakes. Hopefully we don't do it too often.
